### ISSUE TITLE
DP3 always starts with a number of threads equal to NCPUs

### ISSUE THREAD

**Issue by aroffringa** [lofar-astron/DP3#96]
Reported face to face by @revoltek: even when explicitly setting the nthreads to 1, DP3 will still spawn a number of threads equal to the nr of cpus in the machine. These threads are not given tasks, so that indeed the %CPU usage is not above the requested number of threads, but it is nevertheless a bit confusing for debugging, and when launching e.g. 64 times a DP3 process with nthreads=1, the 4000 threads that are spawned do take some resources (even though it does not seem to be an issue right now).

This is probably a small change, like having to construct the thread pools _after_ the nthreads is own instead of before.

**Comment by revoltek** [comment]
Maybe I am checking it in the wrong way, I see e.g. on htop that before running DPPP I've 17 threads alive 1 running (htop). After starting DPPP (with numthreads = 1) I've 87 threads and 2 runnings (DPPP+htop).

I tested it with:
msin = VirgoA_SB294.MS
msin.datacolumn = DATA
msout = .
msout.datacolumn = CORRECTED_DATA
numthreads = 1
steps = [sol]
sol.type = ddecal
sol.uvlambdamin = 30
sol.mode = rotation+diagonal
sol.maxiter = 500
sol.tolerance = 1e-3
sol.nchan = 1
sol.solint = 1
sol.propagatesolutions = True
sol.usemodelcolumn = True
sol.minvisratio=0.6

![screenshot_20190117_121201](https://user-images.githubusercontent.com/4458741/51314840-0defc000-1a51-11e9-8373-1ce900e70774.png)

**Comment by aroffringa** [comment]
I ran your parset and see only one thread started, so I can't reproduce this issue. I also checked the thread pool structures again and don't see issues in DP3.

I think it might be because of different versions of e.g. lapack. Since it is only an issue with reporting and not with actual performance (i.e., numthreads=1 does use only one thread even for @revoltek ) I think this is fine, and I'll close the issue.
